### ISSUE TITLE
NVDA should take one back to the previous position on a page such as a link

### ISSUE THREAD

**Issue by nvdaTrac** [nvaccess/test_nvdaTracImport#212]
**Reported by mike reiser on 8 Jul 2008 20:38 UTC**
Currently, NVDA does not remember the place someone leaves off at when going back to the previous page.  Steps to reproduce:

1.  Go to a site such as google and do a search.  

2.  Follow a link of interest.  

3.  Press alt leftarrow or backspace to go to the previous page.  

Expected results: NVDA should go back to the link the person was at when they activated it.  

Actual results: NVDA goes back to the top of the page and looses place of where the person was at on the previous page.  I see this only being applicable to links as one does not usually go back to previous text IE headings or paragraphs.

**Comment by nvdaTrac** [comment]
**Comment by jteh on 9 Jul 2008 01:20 UTC**
Are you referring to the new virtual buffers used in Firefox 3? NVDA does return to the correct link when I move back in Firefox 3 following your Google search example.

I have seen sites where this does not work properly. Please note that NVDA does not remember previous pages at all. It relies on the browser to correctly position the focus on the link followed to reach the next page. If this focus is not set properly for some reason, this will break. This seems to happen with Trac. I suspect Trac may be using some javascript to focus elsewhere or perhaps it is a browser bug. However, it certainly behaves correctly in the Google example for me.

**Comment by nvdaTrac** [comment]
**Comment by jteh on 9 Jul 2008 01:37 UTC**
There definitely seems to be some sort of javascript used by Trac which messes with the focus when returning to some pages. With javascript enabled, the focus gets set to the document when I press back. However, if I disable javascript, the focus is correctly returned to the link. I'm not sure if this is a Trac bug or a Firefox bug. :)

**Comment by nvdaTrac** [comment]
**Comment by mike.reiser on 9 Jul 2008 14:14 UTC**
Replying to [jteh](comment:1):
> Are you referring to the new virtual buffers used in Firefox 3? NVDA does return to the correct link when I move back in Firefox 3 following your Google search example.
> 
> I have seen sites where this does not work properly. Please note that NVDA does not remember previous pages at all. It relies on the browser to correctly position the focus on the link followed to reach the next page. If this focus is not set properly for some reason, this will break. This seems to happen with Trac. I suspect Trac may be using some javascript to focus elsewhere or perhaps it is a browser bug. However, it certainly behaves correctly in the Google example for me.
Interesting, I just did a google search for shark.  I followed a link to a wikipedia article, and when I went back it took me to the top and not the link I was on.  I am refering to firefox3 and am using r2204.  This happens for me consistantly on all pages.

**Comment by nvdaTrac** [comment]
**Comment by mike.reiser on 9 Jul 2008 15:54 UTC**
Replying to [jteh](comment:2):
> There definitely seems to be some sort of javascript used by Trac which messes with the focus when returning to some pages. With javascript enabled, the focus gets set to the document when I press back. However, if I disable javascript, the focus is correctly returned to the link. I'm not sure if this is a Trac bug or a Firefox bug. :)
Just tested with both java and java script disabled, still the same results on all pages.  Did another google search and still the same results.  Using R 2206 source version and firefox3.

**Comment by nvdaTrac** [comment]
**Comment by Big.D on 9 Aug 2008 23:40 UTC**
Yeah I've also tried this with <a href="http://www.zonebbs.com">The Zone BBS</a> and it doesn't return to the position I was on when I left that page either. Using r2343.

Thanks.

**Comment by nvdaTrac** [comment]
**Comment by jteh on 11 Aug 2008 23:07 UTC**
Replying to [Big.D](comment:5):
> Yeah I've also tried this with The Zone BBS and it doesn't return to the position I was on when I left that page either. Using r2343.
It doesn't work for me there either. However, does it work for you on other sites; e.g. the Google example I provided above?

**Comment by nvdaTrac** [comment]
**Comment by Big.D on 11 Aug 2008 23:56 UTC**
Hi,

I did a google search for ESpeak varients and went to the second search result, then pressed backspace. Both with screen layout on and off, it didn't work.

Thanks.

**Comment by nvdaTrac** [comment]
**Comment by jteh on 26 Aug 2008 04:16 UTC**
I have done some more testing and can confirm that this is broken for all pages in Firefox 3.0.1. I'm not sure about 3.0. This seems to work fine in Firefox 3.0.2 and 3.1 nightlies, so it'd be good if someone having this problem could test with a nightly build of either 3.0.2 or 3.1.

I couldn't find a bug report indicating that this is a fixed regression in 3.0.2, but this does seem to be the case. Marco, are you aware of any such bug?

**Comment by nvdaTrac** [comment]
**Comment by jteh on 26 Aug 2008 04:18 UTC**
Clarification: This is still broken for me for Trac and www.zonebbs.com, but it works fine on most other pages, including the Google search example above.

**Comment by nvdaTrac** [comment]
**Comment by jteh on 16 Oct 2008 22:18 UTC**
Are those experiencing this bug running !WebVisum? Can you please try the Google search scenario with !WebVisum disabled? You can disable !WebVisum from the Add-ons dialog. Make sure you restart Firefox after disabling it. Our testing shows that there appears to be an issue with !WebVisum and returning to the correct position on a page. This obviously needs to be resolved, but I'd like to know if anyone else can confirm this.
